app/services/legacy/dataset_import_service.rb

```diff
@@ -32,7 +32,7 @@ def dataset_attributes
       frequency: build_frequency,
       published_date: legacy_dataset["metadata_created"],
       created_at: legacy_dataset["metadata_created"],
-      last_updated_at: legacy_dataset["metadata_modified"],
```

Comment:
The Find application is dependent on `last_updated_at` being present:

https://github.com/alphagov/datagovuk_find/blob/201a714970e1b35c78960606588de1bb6c1b6641/app/services/search/query.rb#L182

Therefore we need to make these changes in a backward compatible way so that we don't require any downtime.

1. Add `public_updated_at` field, with `last_updated_at` being an alias. Release and re-index.
2. Release a new version of find that now uses `public_updated_at`.
3. Remove `last_updated_at` from Publish,  release and re-index.